lisa/sut_orchestrator/hyperv/platform_.py

```diff
@@ -0,0 +1,323 @@
+from lisa.util.logger import Logger, get_logger
+
+from .. import HYPERV
+from .context import get_node_context
+from .schema import HypervNodeSchema, HypervPlatformSchema
+from .serial_console import SerialConsole, SerialConsoleLogger
+
+
+class _HostCapabilities:
+    def __init__(self) -> None:
+        self.core_count = 0
+        self.free_memory_mib = 0
+
+
+class HypervPlatform(Platform):
+    def __init__(
+        self,
+        runbook: schema.Platform,
+    ) -> None:
+        super().__init__(runbook=runbook)
+
+    @classmethod
+    def type_name(cls) -> str:
+        return HYPERV
+
+    @classmethod
+    def supported_features(cls) -> List[Type[feature.Feature]]:
+        return [SerialConsole]
+
+    def _initialize(self, *args: Any, **kwargs: Any) -> None:
+        self._hyperv_runbook = self._get_hyperv_runbook()
+        self.server_node = self._initialize_server_node()
+        self._host_capabilities = self._init_host_capabilities(self._log)
+        self._common_vhd_path: Optional[PureWindowsPath] = None
+
+    def _get_hyperv_runbook(self) -> HypervPlatformSchema:
+        hyperv_runbook = self.runbook.get_extended_runbook(HypervPlatformSchema)
+        assert hyperv_runbook, "platform runbook cannot be empty"
+        return cast(HypervPlatformSchema, hyperv_runbook)
+
+    def _initialize_server_node(self) -> RemoteNode:
+        assert self._hyperv_runbook, "hyperv runbook cannot be empty"
+
+        if len(self._hyperv_runbook.servers) > 1:
+            self._log.info(
+                "Multiple servers are currently not supported. "
+                "Only the first server will be used."
+            )
+
+        server = self._hyperv_runbook.servers[0]
+        server_node = RemoteNode(
+            runbook=schema.Node(name="hyperv"),
+            index=-1,
+            logger_name="hyperv",
+            parent_logger=get_logger("hyperv"),
+        )
+
+        server_node.set_connection_info(
+            address=server.address, username=server.username, password=server.password
+        )
+
+        return server_node
+
+    def _prepare_environment(self, environment: Environment, log: Logger) -> bool:
+        return self._configure_node_capabilities(environment, log)
+
+    def _configure_node_capabilities(
+        self, environment: Environment, log: Logger
+    ) -> bool:
+        if not environment.runbook.nodes_requirement:
+            return True
+
+        nodes_capabilities = self._create_node_capabilities()
+
+        nodes_requirement = []
+        for node_space in environment.runbook.nodes_requirement:
+            if not node_space.check(nodes_capabilities):
+                return False
+
+            requirement = node_space.generate_min_capability(nodes_capabilities)
+            nodes_requirement.append(requirement)
+
+        environment.runbook.nodes_requirement = nodes_requirement
+        return True
+
+    def _init_host_capabilities(self, log: Logger) -> _HostCapabilities:
```

Comment:
`_init_host_capabilities` should be `_get_host_capabilities`, because it returns capability, and not save somewhere.

Reply:
renamed.